plugins/modules/cm_resource.py

```diff
@@ -0,0 +1,156 @@
+description:
+  - Create, update, and delete resources from ad-hoc Cloudera Manager API endpoint paths, i.e. unimplemented API calls.
+  - This module only supports the C(POST), C(PUT), and C(DELETE) HTTP methods.
+  - To retrieve details, i.e. read-only, from ad-hoc/unimplemented API endpoints, use the M(cloudera.cluster.cm_resource_info) module.
+  - The module supports C(check_mode).
+author:
+  - "Webster Mudge (@wmudge)"
+requirements:
+  - cm_client
+options:
+  method:
+    description:
+      - HTTP method for the CM API endpoint path.
+    type: str
+    required: True
+    choices:
+        - DELETE
+        - POST
+        - PUT
+  body:
+    description:
+      - HTTP body for the CM API endpoint call.
+    type: dict
+extends_documentation_fragment:
+  - cloudera.cluster.cm_options
+  - cloudera.cluster.cm_endpoint
+  - cloudera.cluster.cm_resource
+'''
+
+EXAMPLES = r'''
+---
+- name: Create a new local Cloudera Manager user
+  cloudera.cluster.cm_resource:
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    path: "/user"
+    method: "POST"
+    body:
+      items:
+        - name: new_user
+          password: "Als*$ecU7e"
+
+- name: Update a Cloudera Manager user
+  cloudera.cluster.cm_resource:
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    path: "/user/existing_user"
+    method: "PUT"
+    body:
+      authRoles:
+        - name: "ROLE_LIMITED"
+        
+- name: Delete a Cloudera Manager user
+  host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    path: "/user/existing_user"
+    method: "DELETE"
+'''
+
+RETURN = r'''
+---
+resources:
+    description:
+        - The results from the Cloudera Manager API endpoint call.
+        - If the I(field) is found on the response object, its contents will be returned.
+    type: list
+    elements: complex
+    returned: always
+'''
+
+class ClouderaResource(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaResource, self).__init__(module)
+        
+        # Set parameters
+        self.method = self._get_param('method')
+        self.path = self._get_param('path')
+        self.query = self._get_param('query', dict())
+        self.field = self._get_param('field')
+        self.body = self._get_param('body')
+        
+        # Initialize the return values
+        self.resources = []
+        
+        # Execute the logic
+        self.process()
+    
+    @ClouderaManagerModule.handle_process
+    def process(self):
+        if not self.module.check_mode:
+            self.resources = self.call_api(self.path, self.method, self.query, 
+                                           self.field, self.body)
+            
+            
+def main():
+    module = ClouderaManagerModule.ansible_module(
+        argument_spec=dict(
+            method=dict(required=True, type='str', choices=['POST', 'PUT', 'DELETE']),
```

Comment:
Does this still allow auth and host to be passed?

Reply:
Yes, the `ansible_module()` method wraps `user`, `password`, `host`, and `endpoint_url` all as valid means of connecting and authenticating. The wrapping allows the use of the "inner" method to be used for a "discovery" mode for the CM endpoint.